sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -184,7 +183,8 @@ abstract class Optimizer(catalogManager: CatalogManager)
       ReplaceIntersectWithSemiJoin,
       ReplaceExceptWithFilter,
       ReplaceExceptWithAntiJoin,
-      ReplaceDistinctWithAggregate) ::
+      ReplaceDistinctWithAggregate,
+      ReplaceDeduplicateWithAggregate) ::
```

Comment:
Why we need move this rule?

Reply:
It needs to put the rule 'CombineUnions' before ReplaceDistinctWithAggregate and ReplaceDeduplicateWithAggregate, otherwise these two rules will replace Distinct/Deduplicate to Aggregate and lead to rule 'CombineUnions' is invalid. On the other hand putting the rule 'ReplaceDeduplicateWithAggregate' in batch 'Replace Operators' is more intuitional.